x-pack/plugins/security_solution/common/detection_engine/schemas/request/patch_rules_schema.ts

```diff
@@ -124,6 +126,8 @@ export const patchRulesSchema = t.exact(
     threat_indicator_path,
     concurrent_searches,
     items_per_search,
+    new_terms_fields: newTermsFields,
+    history_window_start: historyWindowStart,
```

Comment:
If defined generally enough (e.g. "context range") we could potentially reuse a field like this for other rules/situations. Not sure if that generalization would be useful; I think my suggestion is mainly a "oh no, more fields" reaction. What do you think?

Reply:
For most rule types I think the "context range" would be defined by `from` and `to`, right? I couldn't think of any other places to apply an additional time range parameter with the same semantics as `history_window_start`. The closest one might be an update to threshold rules to allow defining the bucket window separately from the rule interval, but the semantics there are different enough that I would favor using a separate field specifically for the bucket size.

Reply:
Yep, that seems reasonable. Thanks!